### PR TITLE
[REEF-480] Too many open file descriptors in NetworkServiceTest runni…

### PR DESCRIPTION
**Pull request by Gyeongin**
…ng in Linux

This addressed the issue by
  * Fixed NetworkMessagingTest.close() to close receiverResolver and senderResolver
  * Used try-with-resoureces

JIRA:
  [REEF-480](https://issues.apache.org/jira/browse/REEF-480)

### CODE REVIEW

File: lang/java/reef-io/src/test/java/org/apache/reef/services/network/NetworkConnectionServiceTest.java [hunk 1]
```diff
@@ -77,10 +77,10 @@ private void runMessagingNetworkConnectionService(Codec<String> codec) throws Ex
       monitor.mwait();
     } catch (NetworkException e) {
       e.printStackTrace();
+    } finally {
```
Comment (markusweimer) [inline lang/java/reef-io/src/test/java/org/apache/reef/services/network/NetworkConnectionServiceTest.java]: Can we use try-with-resources here?

File: lang/java/reef-io/src/test/java/org/apache/reef/services/network/NetworkConnectionServiceTest.java [hunk 2]
```diff
@@ -114,48 +114,50 @@ public void runNetworkConnServiceWithMultipleConnFactories(Codec<String> stringC
...
-          conn.open();
-          for (int count = 0; count < groupcommMessages; ++count) {
-            // send messages to the receiver.
-            conn.write("hello" + count);
+    try {
+      executor.submit(new Runnable() {
+        @Override
+        public void run() {
+          final Connection<String> conn = messagingTest.getConnectionFromSenderToReceiver(groupCommClientId);
+          try {
```
Comment (markusweimer) [inline lang/java/reef-io/src/test/java/org/apache/reef/services/network/NetworkConnectionServiceTest.java]: Same here, try-with-resources should make sure we never forget to close.

File: lang/java/reef-io/src/test/java/org/apache/reef/services/network/NetworkConnectionServiceTest.java [hunk 3]
```diff
@@ -114,48 +114,50 @@ public void runNetworkConnServiceWithMultipleConnFactories(Codec<String> stringC
...
-          conn.open();
-          for (int count = 0; count < shuffleMessges; ++count) {
-            // send messages to the receiver.
-            conn.write(count);
+      executor.submit(new Runnable() {
+        @Override
+        public void run() {
+          final Connection<Integer> conn = messagingTest.getConnectionFromSenderToReceiver(shuffleClientId);
+          try {
+            conn.open();
```
Comment (markusweimer) [inline lang/java/reef-io/src/test/java/org/apache/reef/services/network/NetworkConnectionServiceTest.java]: Same here, try-with-resources should make sure we never forget to close.

**Comment by markusweimer** [comment]
I did a quick pass. The change would benefit from more use of try-with-resources, beyond the places I commented on. Also, there some purely cosmetic indention changes we should minimize before merge.

**Comment by Gyeongin** [comment]
Thank you.
Since our resources are created in NetworkMessagingTest, it is hard to make them closed by using try-with-resources in NetworkConnectionServiceTest.
To do this, we shoud make NetworkMessagingTest class implement autoclosable, but I think this change is not good in our case.

**Comment by Gyeongin** [comment]
I changed my commit to use try-with-resources.
Now NetworkMessagingTest implements Autocloseable.

File: lang/java/reef-io/src/test/java/org/apache/reef/services/network/NetworkConnectionServiceTest.java [hunk 4]
```diff
@@ -64,23 +64,22 @@ public NetworkConnectionServiceTest() throws InjectionException {
...
+      try (final Connection<String> conn = messagingTestService.getConnectionFromSenderToReceiver(groupCommClientId)) {
+        try {
+          conn.open();
+          for (int count = 0; count < numMessages; ++count) {
+            // send messages to the receiver.
+            conn.write("hello" + count);
+          }
+          monitor.mwait();
+        } catch (NetworkException e) {
+          e.printStackTrace();
```
Comment (markusweimer) [inline lang/java/reef-io/src/test/java/org/apache/reef/services/network/NetworkConnectionServiceTest.java]: Shouldn't an exception here indicate test failure?

File: lang/java/reef-io/src/test/java/org/apache/reef/services/network/NetworkConnectionServiceTest.java [hunk 5]
```diff
@@ -188,33 +183,35 @@ public void testMessagingNetworkConnServiceRate() throws Exception {
...
+          long start = System.currentTimeMillis();
+          try {
+            conn.open();
+            for (int count = 0; count < numMessages; ++count) {
+              // send messages to the receiver.
+              conn.write(message);
+            }
+            monitor.mwait();
+          } catch (NetworkException e) {
+            e.printStackTrace();
```
Comment (markusweimer) [inline lang/java/reef-io/src/test/java/org/apache/reef/services/network/NetworkConnectionServiceTest.java]: Same as above: Is it safe to swallow the exception? If so, please document why.

File: lang/java/reef-io/src/test/java/org/apache/reef/services/network/NetworkConnectionServiceTest.java [hunk 6]
```diff
@@ -237,31 +234,31 @@ public void testMessagingNetworkConnServiceRateDisjoint() throws Exception {
...
+
+              try {
+                conn.open();
+                for (int count = 0; count < numMessages; ++count) {
+                  // send messages to the receiver.
+                  conn.write(message);
+                }
+                monitor.mwait();
+              } catch (NetworkException e) {
+                e.printStackTrace();
```
Comment (markusweimer) [inline lang/java/reef-io/src/test/java/org/apache/reef/services/network/NetworkConnectionServiceTest.java]: Same as above

**Comment by markusweimer** [comment]
@Gyeongin Thanks, I did another pass.

**Comment by taegeonum** [comment]
The build failures Reef-pull-request-ubuntu #134, Reef-pull-request-ubuntu #135  in ubuntu are because of wrong configuration in Jenkins. I've fixed the configuration, and now it will work.

**Comment by Gyeongin** [comment]
@taegeonum Thanks.

**Comment by Gyeongin** [comment]
@markusweimer I think your comment is valid, and changed to throw exception.
Also, I applied these changes to all tests in reef-io/src/test/java/org/apache/reef/services/network/

**Comment by markusweimer** [comment]
@Gyeongin I'll test and merge.

**Pull request closed without merging**